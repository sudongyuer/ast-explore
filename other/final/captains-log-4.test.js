import pluginTester from 'babel-plugin-tester'
import captainsLog from './captains-log-4'

pluginTester({
  plugin: captainsLog,
  snapshot: true,
  tests: [
    {code: `anything.log();`, snapshot: false},
    `console.log('sup dawg')`,
    `
      function add(a, b) {
        console.log(a, b)
        return a + b
      }
    `,
    `
      const multiply = (a, b) => {
        console.log(a, b)
        return a * b
      }
    `,
    `
      const divide = function(a, b) {
        console.log(a, b)
        return a / b
      }
    `,
  ],
})
