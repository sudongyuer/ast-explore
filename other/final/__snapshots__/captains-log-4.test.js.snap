// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2. captains-log 1`] = `
"
console.log('sup dawg')

      ↓ ↓ ↓ ↓ ↓ ↓

console.log('1:0', 'sup dawg');
"
`;

exports[`3. captains-log 1`] = `
"
function add(a, b) {
  console.log(a, b)
  return a + b
}

      ↓ ↓ ↓ ↓ ↓ ↓

function add(a, b) {
  console.log(\\"add 2:2\\", a, b);
  return a + b;
}
"
`;

exports[`4. captains-log 1`] = `
"
const multiply = (a, b) => {
  console.log(a, b)
  return a * b
}

      ↓ ↓ ↓ ↓ ↓ ↓

const multiply = (a, b) => {
  console.log(\\"multiply 2:2\\", a, b);
  return a * b;
};
"
`;

exports[`5. captains-log 1`] = `
"
const divide = function(a, b) {
  console.log(a, b)
  return a / b
}

      ↓ ↓ ↓ ↓ ↓ ↓

const divide = function (a, b) {
  console.log(\\"divide 2:2\\", a, b);
  return a / b;
};
"
`;
